lib/src/model/path_utils.dart

```diff
@@ -0,0 +1,36 @@
+import 'dart:ui';
+
+class RingNumberProvider<T> {
```

Comment:
I'd like to avoid generics where not needed - please remove `<T>`

Reply:
I thought this ring-buffer-ish class clould be useful in future for other things as well. Thats why I made it generic. 
Do you think thats not needed in future?

Reply:
If it's needed we can add it then. No need to complicate the code now for something that we don't need.